src/models/data.py

```diff
@@ -130,34 +148,73 @@ def __init__(self, loader: DataLoader) -> None:
         self.shuffle = loader.shuffle
         self.clear_nans = loader.clear_nans
         self.to_tensor = loader.to_tensor
+        self.experiment = loader.experiment
 
         if self.shuffle:
             # makes sure they are shuffled every epoch
             shuffle(self.data_files)
 
         self.normalizing_dict = loader.normalizing_dict
-        self.normalizing_array: Optional[Dict[str, np.ndarray]] = None
+        self.historical_normalizing_array: Optional[Dict[str, np.ndarray]] = None
+        self.current_normalizing_array: Optional[Dict[str, np.ndarray]] = None
 
         self.idx = 0
         self.max_idx = len(loader.data_files)
 
     def __iter__(self):
         return self
 
-    def calculate_normalizing_array(self, data_vars: List[str]):
+    def calculate_normalizing_array(self, data_vars: List[str], target_var: str):
         # If we've made it here, normalizing_dict is definitely not None
         self.normalizing_dict = cast(Dict[str, Dict[str, np.ndarray]], self.normalizing_dict)
 
-        mean, std = [], []
-        for var in data_vars:
-            mean.append(self.normalizing_dict[var]['mean'])
-            std.append(self.normalizing_dict[var]['std'])
+        if self.experiment == 'nowcast':
+            historical_mean, historical_std = [], []
+            current_mean, current_std = [], []
+            for var in data_vars:
+                # historical_mean = all times up to current time
+                historical_mean.append(self.normalizing_dict[var]['mean'][:-1])
+                historical_std.append(self.normalizing_dict[var]['std'][:-1])
+
+                # current_mean = the current times for all EXCEPT for target_var
+                if var != target_var:
```

Comment:
this is the right way to do it no? Because we will always have more predictor variables (Precip, Soil Moisture etc.) and only ever have one predictand (VHI)

Reply:
or have i misunderstood

Reply:
Some of the input data is not forecast data, thought.

For example, the CHIRPS data is not available into the future. This makes it a little misleading to use it in the nowcasting model, since it wouldn't be usable in a "future prediction" scenario.

I guess the question is whether the nowcasting model's "current" data should only have forecasted data (e.g. SEAS5 stuff) or whether we just want to test the model's ability to nowcast generally.

Reply:
i think the current data can also have historical data (i.e. we're testing the scenario of we know the rainfall and soil moisture in this time, how can we convert these numbers into vegetation health without a physical process model?)

this will also allow us to train the model on HISTORICAL relationships between precip now -> VHI now because there might not be enough data to just use the S5 forecasts.

Reply:
right, that makes sense